### DISCUSSION TITLE
Feature request: PlayerEvents for VolUp and VolDown

### DISCUSSION THREAD

**Discussion by Muffexx** [librespot-org/librespot#618] [Feature Requests]
Hi,
since there are PlayerEvents (player.rs) for **started**, **changed** and **stopped**, would it be possible to add events for **VolUp** and **VolDown?** This way I could control the volume of my receiver by running scripts that use HDMI-CEC (https://github.com/Pulse-Eight/libcec).

Even better would be to directly integrate HDMI-CEC-support, but I assume this would consume too much time.

Thanks for your reply.

Kind regards

**Comment by michaelherger** [comment]
I have something similar added to my branch: https://github.com/michaelherger/librespot/commit/fb7b8d7230d0c9f3ccc862801297d5ab575b994a. It would trigger a volume change event with the actual value sent to the event handler. Ignore the `self.lms.is_configured` condition - that's specific to my application. And the `lms.rs` is my custom event handler implementation. That change would belong in `player_event_handler.rs`.

**Comment by delMar43** [comment]
> Even better would be to directly integrate HDMI-CEC-support, but I assume this would consume too much time.

I like the idea of just exposing the event. That leaves me with the possibility to either change volume of the AVR via HDMI-CEC, or via API (in case I decide to connect via TOS-Link or analog, instead of HDMI)

Alongside the new volume-change events, we'd probably also need a way to suppress volume changes of the internal player, as that volume should stay at eg 75 or 100%.

In case it's relevant, I'd like to vote for merging this.

**Comment by Muffexx** [comment]
> Alongside the new volume-change events, we'd probably also need a way to **suppress volume changes of the internal player**, as that volume should stay at eg 75 or 100%.

This would be even better.
